index.Rmd

```diff
@@ -131,335 +174,248 @@ In its most basic use-case, `read_csv` expects that the data file:
-scatter plot to represent each language's higher-level language category.
+Learning how to describe data visualizations is a very useful skill. We will provide descriptions for you in this course (as we did above) until we get to Chapter 4, which focuses on data visualization. Then, we will explicitly teach you how to do this yourself and not over-state or over-interpret the results from a visualization.
 
-To do this, we modify our scatter plot code above. Specifically, we will add an argument to the `aes` function, specifying that the points should be coloured by the `category` column:
+### Putting it all together
 
-```{r scatter-colour-by-category, fig.width=7.75, fig.height=4, warning=FALSE,  fig.cap = "Scatter plot of proportion of Canadians reporting a language as their mother tongue vs the primary language at home coloured by language category"}
-ggplot(can_lang, aes(x = most_at_home, y = mother_tongue, color = category)) +
-  geom_point() +
-  xlab("Language spoken most at home (proportion of Canadian residents)") +
-  ylab("Mother tongue (proportion of Canadian residents)") +
-  scale_x_log10(labels = scales::comma) +
-  scale_y_log10(labels = scales::comma)
-```
+In the code chunk below, we put everything from this chapter together. We have added a few more layers to make the data visualization even more effective. Specifically, we changed the colour of the bars and changed the background from grey to white to improve the contrast. Note that we actually skipped the `select` step that we did above just to show you that we didn't actually need to `select` the columns to create the visualization! 
 
-What do we see when considering the second part of our exploratory question? Do we see a difference in the pattern of the relationship between the number of people who speak a language as their mother tongue and the number of people who speak a language as their primary spoken language at home between higher-level language categories? Probably not!
+> Notice that we provided *comments* beside some of the code below using the hashtag symbol `#`. You can use comments to explain lines of code for others or yourself in the future! If R sees a `#` sign, it will ignore all the code that comes after it on that line. It's good practice to get in the habit of commenting your code to improve your code's readability.
 
-For each higher-level language category there appears to be a positive relationship between the number of people who speak a language as their mother tongue and the number of people who speak a language as their primary spoken language at home. This relationship looks similar, regardless of the category. 
+```{r nachos-to-cheesecake, fig.width=7.75, fig.height=4, warning=FALSE, message=FALSE, fig.cap = "Putting it all together: Bar plot of the ten Aboriginal languages most often reported by Canadians as their mother tongue"}
+library(tidyverse)
```

Comment:
There is too much whitespace between lines in the code here. I would do it like this instead:

```
library(tidyverse)

can_lang <- read_csv("data/can_lang.csv")

aboriginal_lang <- filter(can_lang, category == "Aboriginal languages")
arranged_lang <- arrange(aboriginal_lang, by = desc(mother_tongue))
ten_lang <- slice(arranged_lang, 1:10)

ggplot(ten_lang, aes(
  x = reorder(language, mother_tongue),
  y = mother_tongue
)) +
  geom_bar(stat = "identity", fill = "steelblue") +
  xlab("Language") +
  ylab("Mother tongue (Number of Canadians residents)") +
  coord_flip() +
  theme_bw() # use a theme to have a white background
```